src/code42cli/click_ext/types.py

```diff
@@ -0,0 +1,39 @@
+import json
+import re
+
+import click
+
+
+class JSONFileOrString(click.File):
+    """Declares a parameter to be either a JSON string or a path to a file containing parsable
+    JSON.
+
+    If the argument's first non-whitespace character is a "{", assumes it's JSON and tries to parse
+    it, otherwise assumes it's a filename and tries to open file then parse contents as JSON.
+    """
+
+    def __init__(self):
+        super().__init__("r")
+
+    def convert(self, value, param, ctx):
+        if re.match(
+            r"\s*\{", value
```

Comment:
`curl` does this by treating arguments that have the first character as `@` as files. Could be a good opportunity to borrow a familiar cli pattern.

Reply:
Yeah, at first I was thinking it would be simpler for the user to just pass a bare filename or query and not make it explicit, but I think it is actually probably better to be.